src/videostreamindex/DefaultVideoStreamIndex.ts

```diff
@@ -67,12 +74,23 @@ export default class DefaultVideoStreamIndex implements VideoStreamIndex {
     this.currentIndex = indexFrame;
     this.streamToAttendeeMap = null;
     this.streamToExternalUserIdMap = null;
-    this.ssrcToStreamMap = null;
+  }
+
+  integrateSubscribeFrame(): void {
+    // This is called just as a Subscribe is being sent.  Save corresponding Index
+    this.indexForSubscribe = this.currentIndex;
```

Comment:
is it possible that a track event comes late (or browser event loop is running late) and track event happens at the next negotiation, index frame is already overwritten ?

Reply:
I think it should be ok.  Track events only occur when a track is added or removed and occurs right after receiving the subscribeAck.  The mapping from stream to attendee will happen just as subscribeAck happens using the indexForSubscribe.  If for some reason the track event never comes, then a tile won't be added and I don't see an issue.  If it comes late, then it is using that mapping that was created when the subscribeAck happened.  I don't think an old one would ever come after a second subscribeAck.

Reply:
> If for some reason the track event never comes, then a tile won't be added and I don't see an issue

in this case, on unified plan, next time negotiation happens, won't track event never get fired and we are missing a tile ? 

> I don't think an old one would ever come after a second subscribeAck.

I guess it's a rare case but want to be sure on the model here. I couldn't find any documentation on the timing of track event. With early media, I used to expect event track come around sdp answer, but from various logs, track event can be really late.